addons/sms/models/mail_notification.py

```diff
@@ -10,12 +11,63 @@ class MailNotification(models.Model):
     notification_type = fields.Selection(selection_add=[
         ('sms', 'SMS')
     ], ondelete={'sms': 'cascade'})
-    sms_id = fields.Many2one('sms.sms', string='SMS', index='btree_not_null', ondelete='set null')
+    sms_id_int = fields.Integer('SMS ID (tech)', index='btree_not_null')
+    # Used to give links on form view without foreign key. In most cases, you want to use sms_id_int or sms_uuid.
+    sms_id = fields.Many2one('sms.sms', string='SMS', store=False, compute='_compute_sms_id')
```

Comment:
Maybe we could update naming to also match the trace one ? Note that "sms_sms_id" was chosen to ease grep compared to just "sms_id" (if you want an explanation).
```suggestion
    sms_sms_id = fields.Many2one('sms.sms', string='SMS', store=False, compute='_compute_sms_sms_id')
```

Reply:
And sms_sms_id wasn't difference wasn't to distinguish sms_id from traces from those in notifications?